test/support/test_config.h

```diff
@@ -154,6 +154,12 @@
 #define _PSTL_TEST_COMPLEX_SINH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
 #define _PSTL_TEST_COMPLEX_TANH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
 
+// oneAPI DPC++ compiler 2024.2.0 and earlier is unable to eliminate a "dead" function call to an undefined function
+// within a sycl kernel which MSVC uses to allow comparisons with literal zero without warning
+#define _PSTL_TEST_COMPARISON_BROKEN                                                                                   \
+    (__cplusplus >= 202002L && _MSVC_LANG >= 202002L && _MSVC_STL_VERSION >= 143 && _MSVC_STL_UPDATE >= 202303L &&     \

```

Comment:
In principle, checking ``_MSVC_STL_UPDATE`` makes the check of ``_MSVC_STL_VERSION`` redundant; though I do not mind if you prefer checking both.

Citing https://devblogs.microsoft.com/cppblog/c17-progress-in-vs-2017-15-5-and-15-6/ :

> The STL now identifies itself with two macros, defined by including any STL header (e.g. <ciso646>). _MSVC_STL_VERSION is defined as 141 and will remain that way for the VS 2017 v141 toolset series. _MSVC_STL_UPDATE is defined as 201709 in VS 2017 15.5, and will be increased as features are added in future updates. (The year-month typically won’t correspond to an update’s release date. Our process for updating this macro is “we’re checking in std::meow() and it’s now October, so we need to change the value”, and depending on branch/merge timing, it might take a month or more to ship.) The value for VS 2017 15.6 is not yet set in stone, but it will be 201711 or greater.

See also https://github.com/microsoft/STL/wiki/Macro-_MSVC_STL_UPDATE ; it does not mention ``_MSVC_STL_VERSION`` directly, but in fact its values are only updated with major Visual Studio versions: `141` for VS 2017, `142` for VS 2019, and `143` for VS 2022.